asv/www/graphdisplay.js

```diff
@@ -385,9 +385,19 @@ $(document).ready(function() {
                 var max_curves = 8;
 
                 for (var k = 0; k < params.length; ++k) {
+                    var param_name = param_names[k]
+                    var param_values = params[k]
                     var item = [];
-                    for (var j = 0; j < params[k].length && (j+1)*count <= max_curves; ++j) {
-                        item.push(j);
+                    if (state_selection['p-'+param_name] !== undefined) {
+                        for (var j = 0; j < param_values.length; ++j) {
+                            if (state_selection['p-'+param_name].includes($.asv.convert_benchmark_param_value(param_values[j]))) {
```

Comment:
`$.asv.convert_benchmark_param_value` does not always return a string, so this construct fails. The function probably should be changed to always return a string.

Reply:
Maybe the parameters should not use `$.asv.convert_benchmark_param_value` at all, and instead just use the raw string value.

Reply:
Good catch, this was broken with non string parameters. Drop the use `$.asv.convert_benchmark_param_value` fix it.
I pushed new version of the fist patch, tested against numpy and mercurial suite.